### PR TITLE
ISPN-14488 REST Task execution with cache

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-14488
https://issues.redhat.com/browse/ISPN-14489

### CODE REVIEW

File: server/rest/src/main/java/org/infinispan/rest/resources/TasksResource.java [hunk 1]
```diff
@@ -46,9 +48,25 @@ public Invocations getInvocations() {
...
+      EmbeddedCacheManager cacheManager = invocationHelper.getRestCacheManager().getInstance().withSubject(request.getSubject());
+      ScriptingManager scriptingManager = SecurityActions.getGlobalComponentRegistry(cacheManager).getComponent(ScriptingManager.class);
+      NettyRestResponse.Builder builder = new NettyRestResponse.Builder();
+      if (!scriptingManager.getScriptNames().contains(taskName)) {
+         throw Log.REST.noSuchScript(taskName);
+      }
+
+      return CompletableFuture.supplyAsync(() -> {
+         String script = Subject.doAs(request.getSubject(), (PrivilegedAction<String>) () -> scriptingManager.getScript(taskName));
+         return builder.entity(script).build();
```
Comment (jabolina) [inline server/rest/src/main/java/org/infinispan/rest/resources/TasksResource.java:66]: should we add a content-type here or unnecessary?

**Review by jabolina — APPROVED** [review]
LGTM

File: server/rest/src/main/java/org/infinispan/rest/resources/TasksResource.java (hunk 1 above)
Comment (tristantarrant) [reply server/rest/src/main/java/org/infinispan/rest/resources/TasksResource.java:66]: probably I should set it to `text/plain`. In theory we could determine it from the script metadata (which is not exposed by the ScriptingManager API).

**Comment by jabolina** [comment]
Merged, thanks @tristantarrant!

**Pull request merged**